src/main/java/seedu/address/commons/core/index/IndexList.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.commons.core.index;
+
+import java.util.ArrayList;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an array of zero-based or one-based indices maintained in descending order.
+ * The indices in the array are distinct.
+ */
+public class IndexList {
+    private final ArrayList<Index> indices;
+
+    /**
+     * IndexList can only be created by calling {@link IndexList#emptyList()}.
+     */
+    private IndexList() {
+        indices = new ArrayList<>();
+    }
+
+    /**
+     * Creates a new {@code IndexList} with an empty list.
+     */
+    public static IndexList emptyList() {
+        return new IndexList();
+    }
+
+    /**
+     * Adds a new Index into the list if the {@code Index} is unique.
+     *
+     * @param index a new index to be added into the list.
+     */
+    public void add(Index index) {
+        if (!indices.contains(index)) {
+            indices.add(index);
+        }
+    }
+
+    /**
+     * Returns the current size of the index list.
+     *
+     * @return integer representing the number of elements in the list.
+     */
+    public int size() {
+        return indices.size();
+    }
+
+    public Index getIndex(int index) {
+        // invalid index
+        if (index < 0 || index >= this.size()) {
+            return null;
+        }
+
+        return indices.get(index);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof IndexList)) {
+            return false;
+        }
+
+        IndexList otherList = (IndexList) other;
+
+        if (otherList.size() != this.size()) {
+            return false;
+        }
+
+        // check equal for every pair of index
+        for (int i = 0; i < this.size(); i++) {
+            if (!this.getIndex(i).equals(otherList.getIndex(i))) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        String lineOfIndices = "";
+        for (Index index: indices) {
+            lineOfIndices += index.toString() + " ";
+        }
```

Comment:
This sounds like exactly the wrong way to use string builder (O(n^2) performance). (But of course in practice it won't matter)

Reply:
yep it's inefficient. I don't really know a more OOP-friendly way to build the string.
Realistically, most users wouldn't use delete for large number of indices.